scitt/create_detatched_signed_statement.py

```diff
@@ -0,0 +1,217 @@
+""" Module for creating a SCITT signed statement with a detached payload"""
+
+import hashlib
+import argparse
+
+from typing import Optional
+
+from hashlib import sha256
+
+from pycose.messages import Sign1Message
+from pycose.headers import Algorithm, KID, ContentType
+from pycose.algorithms import Es256
+from pycose.keys.curves import P256
+from pycose.keys.keyparam import KpKty, EC2KpD, EC2KpX, EC2KpY, KpKeyOps, EC2KpCurve
+from pycose.keys.keytype import KtyEC2
+from pycose.keys.keyops import SignOp, VerifyOp
+from pycose.keys import CoseKey
+
+from ecdsa import SigningKey, VerifyingKey
+
+
+# Feed header label comes from version 2 of the scitt architecture document
+# https://www.ietf.org/archive/id/draft-birkholz-scitt-architecture-02.html#name-envelope-and-claim-format
+HEADER_LABEL_FEED = 392
+
+# CWT header label comes from version 4 of the scitt architecture document
+# https://www.ietf.org/archive/id/draft-ietf-scitt-architecture-04.html#name-issuer-identity
+HEADER_LABEL_CWT = 13
+
+# Various CWT header labels come from:
+# https://www.rfc-editor.org/rfc/rfc8392.html#section-3.1
+HEADER_LABEL_CWT_ISSUER = 1
+HEADER_LABEL_CWT_SUBJECT = 2
+
+# CWT CNF header labels come from:
+# https://datatracker.ietf.org/doc/html/rfc8747#name-confirmation-claim
+HEADER_LABEL_CWT_CNF = 8
+HEADER_LABEL_CNF_COSE_KEY = 1
+
+
+# Signed Hash envelope header labels from:
+# https://github.com/OR13/draft-steele-cose-hash-envelope/blob/main/draft-steele-cose-hash-envelope.md
+HEADER_LABEL_PAYLOAD_HASH_ALGORITHM = 998
+HEADER_LABEL_LOCATION_HINT = 999
+
+
+def open_signing_key(key_file: str) -> SigningKey:
+    """
+    opens the signing key from the key file.
+    NOTE: the signing key is expected to be a P-256 ecdsa key in PEM format.
+    """
+    with open(key_file, encoding="UTF-8") as file:
+        signing_key = SigningKey.from_pem(file.read(), hashlib.sha256)
+        return signing_key
+
+
+def open_payload(payload_file: str) -> str:
+    """
+    opens the payload from the payload file.
+    """
+    with open(payload_file, encoding="UTF-8") as file:
+        return file.read()
+
+
+def create_signed_statement(
+    signing_key: SigningKey,
+    payload: str,
+    feed: str,
+    issuer: str,
+    content_type: str,
+    location_hint: str,
+) -> bytes:
+    """
+    creates a signed statement, given the signing_key, payload, feed and issuer
+    """
+
+    # NOTE: for the sample an ecdsa P256 key is used
+    verifying_key: Optional[VerifyingKey] = signing_key.verifying_key
+    assert verifying_key is not None
+
+    # pub key is the x and y parts concatenated
+    xy_parts = verifying_key.to_string()
+
+    # ecdsa P256 is 64 bytes
+    x_part = xy_parts[0:32]
+    y_part = xy_parts[32:64]
+
+    # create a protected header where
+    #  the verification key is attached to the cwt claims
+    protected_header = {
+        Algorithm: Es256,
+        KID: b"testkey",
+        ContentType: content_type,
+        HEADER_LABEL_FEED: feed,
```

Comment:
```suggestion
```

Reply:
As per above comments, we still need this.

Reply:
removed as we use cwt.subject in next release